/* Copyright 2023 The MediaPipe Authors.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
==============================================================================*/

syntax = "proto3";

package mediapipe.tasks.vision.image_generator.proto;

import "mediapipe/framework/calculator.proto";
import "mediapipe/tasks/cc/core/proto/base_options.proto";
import "mediapipe/tasks/cc/vision/image_generator/proto/conditioned_image_graph_options.proto";

option java_package = "com.google.mediapipe.tasks.vision.imagegenerator.proto";
option java_outer_classname = "ControlPluginGraphOptionsProto";

message ControlPluginGraphOptions {
  // The base options for the control plugin model.
  core.proto.BaseOptions base_options = 1;

  // The options for the ConditionedImageGraphOptions to generate control plugin
  // model input image.
  proto.ConditionedImageGraphOptions conditioned_image_graph_options = 2;
}
